### PR TITLE
Maintenance week updates

### PR DESCRIPTION
**Pull request by ehanson8**
### What does this PR do?
Updates app according to our [maintenance week documentation](https://mitlibraries.atlassian.net/wiki/spaces/IN/pages/3445948428/DataEng+Application+Maintenance)

### Helpful background context
This app is used by DDC staff for local processing of top container data and does not have any of the AWS/Docker-related code typically found in our apps. I do not foresee this app ever moving into AWS based on the current usage.

### How can a reviewer manually see the effects of these changes?
Run `make test` and `make lint` to confirm they still pass

### Includes new or updated dependencies?
YES

### Changes expectations for external applications?
NO

### Developer
- [x] All new ENV is documented in README
- [x] All new ENV has been added to staging and production environments
- [x] All related Jira tickets are linked in commit message(s)
- [x] Stakeholder approval has been confirmed (or is not needed)

### Code Reviewer(s)
- [ ] The commit message is clear and follows our guidelines (not just this PR message)
- [ ] There are appropriate tests covering any new functionality
- [ ] The provided documentation is sufficient for understanding any new functionality introduced
- [ ] Any manual tests have been performed **or** provided examples verified
- [ ] New dependencies are appropriate or there were no changes

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -45,29 +43,26 @@ def mock_archivesspace():
...
-        "DEV_USER": "user",
-        "DEV_PASSWORD": "password",
-        "DEV_URL": "mock://example-dev.com",
-        "PROD_USER": "user",
-        "PROD_PASSWORD": "password",
-        "PROD_URL": "mock://example-prod.com",
-    }
-    yield
+def _test_env(monkeypatch):
+    monkeypatch.setenv("DEV_USER", "user")
```
Comment (ghukill) [inline tests/conftest.py:53]: +1 to `monkeypatch`!

**Review by ghukill — APPROVED** [review]
Looks good to me!  Thanks for all the updates.  Looks like a clean update, with fairly minimal changes to code, mostly just (still important) scaffolding.

**Review by jonavellecuerdo — CHANGES_REQUESTED** [review]
I reviewed the updates made via `Files changed`, and I think they look good. I think it would be good if the first section of the README, under the application's title, if we had a paragraph describing at high-level (1) what the application does and (2) what it means to "locally process top container data". 🤔 

I wanted to understand what the application is used for, but I personally couldn't gather its purpose from the instructions written in the README. Not expecting a complete overhaul of the README, but a short description at the beginning would be helpful (and perhaps the added context would make it easier for a new developer to work on the repo should the need arise)! 😄

**Comment by ehanson8** [comment]
Good call @jonavellecuerdo, just added a general description to `README.md`, let me know more detail would be useful!

**Review by jonavellecuerdo — APPROVED** [review]

**Pull request merged**